### PR TITLE
Stop using graphbuilder in NewStreamingMap and NewStreamingSet

### PR DESCRIPTION
**Pull request by willhite**

### CODE REVIEW

File: go/types/map.go [hunk 1]
```diff
@@ -38,22 +38,34 @@ func NewMap(kv ...Value) Map {
...
 func NewStreamingMap(vrw ValueReadWriter, kvs <-chan Value) <-chan Map {
 	var k Value
 	outChan := make(chan Map, 1)
 	go func() {
 		defer close(outChan)
-		gb := NewGraphBuilder(vrw, MapKind, false)
+		ch := newEmptyMapSequenceChunker(vrw, vrw)
+		var lastK Value
 		for v := range kvs {
 			if k == nil {
```
Comment (rafael-atticlabs) [inline go/types/map.go]: let's just be a bit more direct here and maintain a local variable, like `nextIsKey` that we toggle each pass.

File: go/types/map.go [hunk 2]
```diff
@@ -38,22 +38,34 @@ func NewMap(kv ...Value) Map {
...
+// graph_builder.go for building collections with values that are not in order.
 func NewStreamingMap(vrw ValueReadWriter, kvs <-chan Value) <-chan Map {
 	var k Value
 	outChan := make(chan Map, 1)
 	go func() {
 		defer close(outChan)
-		gb := NewGraphBuilder(vrw, MapKind, false)
+		ch := newEmptyMapSequenceChunker(vrw, vrw)
+		var lastK Value
 		for v := range kvs {
```
Comment (rafael-atticlabs) [inline go/types/map.go]: d.PanicIfNil(v)

File: go/types/map.go [hunk 3]
```diff
@@ -38,22 +38,34 @@ func NewMap(kv ...Value) Map {
...
 			if k == nil {
 				k = v
+				if lastK != nil {
```
Comment (rafael-atticlabs) [inline go/types/map.go]: d.PanicIfFalse(lastK == nil || lastK.Less(k))

File: go/types/set.go [hunk 4]
```diff
@@ -31,15 +32,27 @@ func NewSet(v ...Value) Set {
...
 func NewStreamingSet(vrw ValueReadWriter, vals <-chan Value) <-chan Set {
 	outChan := make(chan Set, 1)
 	go func() {
 		defer close(outChan)
-		gb := NewGraphBuilder(vrw, SetKind, false)
+		ch := newEmptySetSequenceChunker(vrw, vrw)
+		var lastV Value
 		for v := range vals {
-			gb.SetInsert(nil, v)
+			if lastV != nil {
```
Comment (rafael-atticlabs) [inline go/types/set.go]: d.PanicIfNil(v)
d.PanicIffalse(lastV == nil || lastV.Less(v))

File: go/types/set.go [hunk 5]
```diff
@@ -31,15 +32,27 @@ func NewSet(v ...Value) Set {
...
+				d.PanicIfFalse(lastV.Less(v))
+			}
+			if v != nil {
```
Comment (rafael-atticlabs) [inline go/types/set.go]: why are we tolerating nil values here?

File: go/types/map.go (hunk 2 above)
Comment (willhite) [inline go/types/map.go]: Done. Had to use `d.PanicIfTrue(v == nil)` to avoid boxing.

File: go/types/map.go (hunk 3 above)
Comment (willhite) [inline go/types/map.go]: Done.

File: go/types/set.go (hunk 4 above)
Comment (willhite) [inline go/types/set.go]: Done.

File: go/types/set.go (hunk 5 above)
Comment (willhite) [inline go/types/set.go]: Good catch. This was leftover from testing.

**Pull request merged**